File: eAutobus.Mobile/eautobusmobile/lib/models/recenzija/Recenzija.dart
class Recenzija {
  String? komentar;
  String? linija;
  int? rasporedVoznjeID;
  int? ocjena;
  DateTime? datumRecenzije;
  String? vrstaUsluga;
  int? kupacID;
  Recenzija(
      {this.komentar,
      this.linija,
      this.ocjena,
      this.datumRecenzije,
      this.rasporedVoznjeID,
      this.vrstaUsluga,
      this.kupacID});

  factory Recenzija.fromJson(Map<String, dynamic> json) {
    return Recenzija(
      komentar: json['komentar'] as String,
      linija: json['rasporedVoznje'] == null
          ? null
          : json['rasporedVoznje'] as String?,
      rasporedVoznjeID: json['rasporedVoznjeID'] as int,
      kupacID: json['kupacID'] as int,
      ocjena: json['ocjena'] as int,
      datumRecenzije: DateTime.parse(json['datumRecenzije'].toString()),
      vrstaUsluga: json['vrstaUsluga'] as String,
    );
  }
  Map<String, dynamic> toJson() => {
        'komentar': komentar,
        'rasporedVoznje': linija,
        "ocjena": ocjena,
        'datumRecenzije': datumRecenzije?.toIso8601String(),
        'raspoderVoznjeID': rasporedVoznjeID,
        'vrstaUsluga': vrstaUsluga,
        'kupacID': kupacID,
      };
}

File: eAutobus/Database/Recenzija.cs
﻿namespace eAutobus.Database
{
    public class Recenzija
    {
        public int RecenzijaID { get; set; }
        public string VrstaUsluga { get; set; }
        public DateTime DatumRecenzije { get; set; }
        public int Ocjena { get; set; }
        public string Komentar { get; set; }
        public int RasporedVoznjeID { get; set; }
        public RasporedVoznje RasporedVoznje { get; set; }
        public int KupacID { get; set; }
        public Kupac Kupac { get; set; }
    }
}
